homeassistant/components/template/light.py

```diff
@@ -391,6 +549,7 @@ async def async_turn_on(self, **kwargs: Any) -> None:
                 run_variables=common_params,
                 context=self._context,
             )
+            self._color_mode = ColorMode.COLOR_TEMP
```

Comment:
This should not be set here, it should be moved to above where the color is optimistically set if there is no template.
```suggestion
```